main.js

```diff
@@ -0,0 +1,284 @@
+                    let matter = docmatter(content.toString('UTF-8'));
+                    let header = yaml.safeLoad(matter.header);
+                    let match_date = re_date.exec(fn);
+
+                    let htBody = Markdown.render(matter.body);
+                    let htMatch = /^(<p>[^]*?<\/p>)/i.exec(htBody.trim());
+                    let date_t = new Date(match_date[0])
+                    return {
+                        body: htBody,
+                        title: header.title,
+                        date: formatDate(date_t),
+                        date_t: date_t,
+                        url: `${dirname}/${fn.replace(/\.md$/i,'.html')}`,
+                        excerpt: htMatch[1]
+                    };
+                })
+            );
+            return Promise.all(promises).then(news => {
+                let obj = {
+                    news: {posts: news.sort((a,b) => Math.sign(b.date_t - a.date_t))}
+                };
+                let finalResult = Object.assign(obj, results)
+                resolve(finalResult);
+            })
+        })
+    });
+}
+
+/**
+ * Writes news.html and news articles from news object
+ * @param {Object} results object including globalContext, news, and template objects
+ * @param {String} dirname folder to write news to
+ */
+function writeNews(results, dirname='news') {
+    // huh, stuff gets real simple when it's sync 🤔
+    fse.mkdirpSync(path.join(outDir, dirname));
+
+    let promises = results.news.posts.map(newsObj =>
+        fse.writeFile(path.join(outDir, newsObj.url), 
+            results.wrapperTemplate(
+                Object.assign({
+                    body: results.articleTemplate({
+                        title: newsObj.title,
+                        date: newsObj.date,
+                        body: newsObj.body
+                    }),
+                    title: newsObj.title,                    
+                }, results.globalContext)
+            )
+        )
+    )
+    promises.push(fse.writeFile(path.join(outDir, 'news.html'),
+        results.wrapperTemplate(Object.assign({
+            title: "News",
+            body: results.newslistTemplate(results.news)
+        }, results.globalContext))
+    ))
+    return Promise.all(promises);
+}
+
+/**
+ * Writes the index page
+ * @param {Object} results object including globalContext, news, and template objects.
+ */
+function writeIndex(results) {
+    return fse.writeFile(path.join(outDir, 'index.html'), results.wrapperTemplate(
+        Object.assign({
+            body: results.indexTemplate(Object.assign({
+                newslist: results.newslistTemplate({
+                    posts: results.news.posts.slice(0,5),
+                    link: true
+                }) // first 5 news articles (if present)
+            }, results.globalContext))
+            // no title, handled by wrapper.handlebars
+        }, results.globalContext)
+    ));
+}
+
+/**
```

Comment:
This function technically needs a description, though I'm willing to overlook it as the function seems clear from title.